lib/vfscore/main.c

```diff
@@ -1318,6 +1318,59 @@ UK_SYSCALL_R_DEFINE(int, rename, const char*, oldpath, const char*, newpath)
 	return -error;
 }
 
+UK_TRACEPOINT(trace_vfs_renameat, "\"%s\" \"%s\"", const char*, const char*);
+UK_TRACEPOINT(trace_vfs_renameat_ret, "");
+UK_TRACEPOINT(trace_vfs_renameat_err, "%d", int);
+
+UK_SYSCALL_R_DEFINE(int, renameat, int, olddirfd, const char*, oldpath, int, newdirfd, const char*, newpath) {
+	char src[PATH_MAX];
+	char dest[PATH_MAX];
+
+	if (!(oldpath[0] == '/' || olddirfd == AT_FDCWD)) {
+		struct vfscore_file *fp;
+		int error = fget(olddirfd, &fp);
+		if (error)
+				return -error;
+
+		struct vnode *vp = fp->f_dentry->d_vnode;
```

Comment:
Codestyle (I know it's a bit archaic, but them's the rules):
- please add a blank line between declarations and code
- please place declarations at the top of a block

(applies to dest block lower down too)